extensions/MYTAditya/CommentPlus.js

```diff
@@ -0,0 +1,376 @@
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentBooleanP6",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "[COLOR] // [COMMENT]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentReporterP7",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "// [COMMENT] [COLOR]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentBooleanP7",
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: "// [COMMENT] [COLOR]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+              COLOR: {
+                type: Scratch.ArgumentType.COLOR,
+                defaultValue: "#037b03",
+              },
+            },
+          },
+          {
+            opcode: "commentCPlusPlus",
+            blockType: Scratch.BlockType.CONDITIONAL,
+            branchCount: 2,
+            text: "// [COMMENT]",
+            arguments: {
+              COMMENT: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "comment",
+              },
+            },
+          },
+        ],
+      };
+    }
+
+    commentHat() {
+      // no-op
+    }
+
+    commentCap(args) {
+      // no-op
+    }
+
+    commentCommand() {
+      // no-op
+    }
+
+    commentC(args, util) {
+      return true;
+    }
+
+    commentReporter(args) {
+      return args.INPUT;
+    }
+
+    commentBoolean(args) {
+      return args.INPUT || false;
+    }
+
+    commentReporterPPP(args) {
+      return args.INPUT;
+    }
+
+    commentBooleanPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentReporterPPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPPPP(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPlus(args) {
+      // no-op
+    }
+
+    commentReporterPlus(args) {
+      // no-op
+    }
+
+    commentCap() {
+      // no-op
+    }
+
+    commentCPlus() {
+      // no-op
+    }
+    
+    commentReporterPlusPlus(args) {
+      return args.INPUT || false;
+    }
+
+    commentBooleanPlusPlus(args) {
+      return args.INPUT || false;
```

Comment:
this is a boolean block, yet it sometimes returns a string and sometimes returns "false" if there's an absence of a value in the input.

apply the same rules as commentReporterPlusPlus, or `Scratch.Cast.toBoolean` the input.